plugins/storage/pyparts/ceph_daemon_checks.py

```diff
@@ -25,6 +25,44 @@ def __init__(self):
         super().__init__()
         self.ceph_osd_df_tree = self.cli.ceph_osd_df_tree()
         self.ceph_versions = self.cli.ceph_versions()
+        self.ceph_report = self.cli.ceph_report()
+        # Remove the last line as that makes it an invalid json.
+        # The extra line comes from the stderr of 'ceph report'.
+        if self.ceph_report and self.ceph_report[-1].startswith('report'):
+            self.ceph_report = self.ceph_report[:-1]
+
+    def check_osdmaps_size(self):
+        """
+        Check if there are too many osdmaps
+        By default mon_min_osdmaps_epochs (=500) osdmaps are stored by the
+        monitors. However, if the cluster isn't healthy for a long time,
+        the number of osdmaps stored will keep increasing which can result
+        in more disk utilization, possibly slower mons, etc.
+
+        Doc: https://docs.ceph.com/en/latest/dev/mon-osdmap-prune/
+        """
+        if not self.ceph_report:
+            return
+
+        try:
+            cr_json = json.loads(''.join(self.ceph_report))
+        except ValueError:
+            return
+
+        try:
+            osdmaps_count = len(cr_json['osdmap_manifest']['pinned_maps'])
+            # The default count is 500. Going above isn't unusual whenever
+            # any recovery/rebalance happens. We arbitrarily chose 5000 as the
+            # "bad" limit (10 x default).
+            if osdmaps_count >= 5000:
+                msg = ("Found {} pinned osdmaps. This can affect mon's"
+                       "performance and also indicate bugs such as "
+                       "https://tracker.ceph.com/issues/44184 and "
+                       "https://tracker.ceph.com/issues/47290"
+                       .format(osdmaps_count))
+                issue_utils.add_issue(issue_types.CephMapsWarning(msg))
+        except Exception:
```

Comment:
Under what conditions would this raise an exception? I'd rather something a bit more explicit/descriptive than a catch-all with no explanation.